Makefile

```diff
@@ -89,47 +87,62 @@ ifneq ($(CHECKSTYLE),0)
 	PERL5LIB=lib/perlcritic:$$PERL5LIB perlcritic lib
 endif
 
-.PHONY: test
-ifeq ($(TRAVIS),true)
-test: run-tests-within-container
-else
-test: checkstyle test-with-database
-endif
-
-.PHONY: test-unit-and-integration
-test-unit-and-integration:
-	OPENQA_CONFIG= prove ${PROVE_LIB_ARGS} ${PROVE_ARGS}
+test-with-db%: 
+	@$(MAKE) $(subst test-with-db,test-with-database,$@)
 
-.PHONY: test-with-database
-test-with-database:
+test-with-db:
+	@$(MAKE) test-with-database   
+ 
+test-with-databas%:
 	test -d $(TEST_PG_PATH) && (pg_ctl -D $(TEST_PG_PATH) -s status >&/dev/null || pg_ctl -D $(TEST_PG_PATH) -s start) || ./t/test_postgresql $(TEST_PG_PATH)
-	$(MAKE) test-unit-and-integration TEST_PG="DBI:Pg:dbname=openqa_test;host=$(TEST_PG_PATH)"
+	$(MAKE) $(subst -with-database,,$@) TEST_PG="DBI:Pg:dbname=openqa_test;host=$(TEST_PG_PATH)"
 	-pg_ctl -D $(TEST_PG_PATH) stop
 
-# prepares running the tests within Docker (eg. pulls os-autoinst) and then runs the tests considering
-# the test matrix environment variables
-# note: This is supposed to run within the Docker container unlike `launch-docker-to-run-tests-within`
-#       which launches the container.
-.PHONY: run-tests-within-container
-run-tests-within-container:
-	script/run-tests-within-container
+test:
+	prove -l -v -r
```

Comment:
this is repeating `$PROVE_ARGS`. Also, how is it supposed to work? For most we need the database setup

Reply:
In my understanding if one needs automatic database setup - they should use `make test-with-database` (exactly what spec files does). Otherwise it is inconsistent is several ways